p2p/protocol/identify/id.go

```diff
@@ -581,6 +586,11 @@ func (ids *idService) createBaseIdentifyResponse(conn network.Conn, snapshot *id
 		}
 		mes.ListenAddrs = append(mes.ListenAddrs, addr.Bytes())
 	}
+	if ids.metricsTracer != nil {
+		ids.metricsTracer.NumProtocols(len(mes.Protocols))
+		ids.metricsTracer.NumAddrs(len(mes.ListenAddrs))
+	}
```

Comment:
we can simplify this a bit if we are fine with counting snaphot.Addrs as Num Address Sent.

Reply:
I'm wondering if we should rework the API a bit. Might be nice to have a `Identify{Push}Received(numProtos, numAddrs int)`. wdyt?

Reply:
agreed. will fix this.